dissect/hypervisor/disk/vmdk.py

```diff
@@ -398,18 +402,69 @@ def __getattr__(self, attr):
         return getattr(self.hdr, attr)
 
 
+RE_EXTENT_DESCRIPTOR = re.compile(
+    r"""
+    ^
+    (?P<access_mode>RW|RDONLY|NOACCESS)\s
+    (?P<sectors>\d+)\s
+    (?P<type>SPARSE|ZERO|FLAT|VMFS|VMFSSPARSE|VMFSRDM|VMFSRAW)
+    (\s?(?P<filename>\".+\")\s?)?
+    (\s?(?P<start_sector>\d+)\s?)?
+    (\s?(?P<partition_uuid>\S+)\s?)?
+    (?P<device_identifier>\S+)?
+    $
+    """,
+    re.VERBOSE,
+)
+
+
+@dataclass
+class ExtentDescriptor:
+    access_mode: str
+    sectors: int
+    type: str
+    filename: str | None = None
+    start_sector: int | None = None
+    partition_uuid: str | None = None
+    device_identifier: str | None = None
+
+    def __post_init__(self) -> None:
+        self._raw = " ".join(map(str, [v for v in self.__dict__.values() if v is not None]))
```

Comment:
I would just pass in `raw` as first argument, makes it much cleaner then reconstructing it (see the note on quote stripping in the other comment).

If you do want it as a synthesized variable, define it something like:
```
raw: str|None = field(default=None, init=False)
```
Also no need to hide it behind a `_`.